test/end-to-end.js

```diff
@@ -970,6 +977,58 @@ describe('end-to-end', function() {
       });
     });
 
+    describe('error handling in dev/prod', function() {
+      it('includes sensitive error details in development mode', function(done) {
+        var app = loopback({ localRegistry: true, loadBuiltinModels: true });
+        var bootOptions = {
+          appRootDir: SANDBOX,
+          env: 'development'
+        };
+        boot(app, bootOptions, function(err) {
+          if (err) return done(err);
+          app.listen(0, function() {
+            request(app)
+              .get('/url-does-not-exist')
+              .expect(404)
+              .end(function(err, res) {
+                app.use(function(err, res) {
```

Comment:
This doesn't work, the handler is never called as can be seen from no `console.log(responseBody)` appearing in the output of `mocha`.